run.py

```diff
@@ -200,8 +200,13 @@ def add_optimizer(self, parameters):
 		Returns an optimizer for learning the parameters of the model
 		
 		"""
-		return torch.optim.Adam(parameters, lr=self.p.lr, weight_decay=self.p.l2)
-
+		if self.p.optimizer == 'Adagrad':
+			return torch.optim.Adagrad(parameters, lr=self.p.lr, weight_decay=self.p.l2)
+		elif self.p.opimizer == 'Adam':
+			return torch.optim.Adam(parameters, lr=self.p.lr, weight_decay=self.p.l2)
+		else:
+			print("Unknown optimizer used! Please use Adam or Adagrad. Aborting!")
+			quit()
```

Comment:
```suggestion
			raise NotImplementedError
```
That's the way it would have been solved for the model the model parameter, but I'm fine with it either way